File: cliente-service/PetCare.Cliente/Models/Clientes/Cliente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PetCareServicios.Models.Clientes
{
    public class Cliente
    {
        [Key]
        public int ClienteID { get; set; }

        [Required]
        public int UsuarioID { get; set; }

        [Required]
        [StringLength(20)]
        public string DocumentoIdentidad { get; set; } = string.Empty;

        [StringLength(15)]
        public string TelefonoEmergencia { get; set; } = string.Empty;

        public bool DocumentoVerificado { get; set; } = false;

        public DateTime? FechaVerificacion { get; set; }

        [StringLength(20)]
        public string Estado { get; set; } = "Activo";

        public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;

        public DateTime? FechaActualizacion { get; set; }

        // Nota: No usamos ForeignKey ni Navigation Property porque estamos usando bases de datos separadas
        // La relación se maneja a nivel de aplicación usando el UsuarioID
    }
} 

File: PetCareSolution-frontend/src/models/Cliente.js
// Modelos para el cliente en el frontend
export class ClienteRequest {
  constructor(documentoIdentidad = '', telefonoEmergencia = '') {
    this.documentoIdentidad = documentoIdentidad;
    this.telefonoEmergencia = telefonoEmergencia;
  }
}

export class ClienteResponse {
  constructor(data = {}) {
    this.clienteID = data.clienteID || 0;
    this.usuarioID = data.usuarioID || 0;
    this.documentoIdentidad = data.documentoIdentidad || '';
    this.telefonoEmergencia = data.telefonoEmergencia || '';
    this.documentoVerificado = data.documentoVerificado || false;
    this.fechaVerificacion = data.fechaVerificacion || null;
    this.estado = data.estado || 'Activo';
    this.fechaCreacion = data.fechaCreacion || null;
    this.fechaActualizacion = data.fechaActualizacion || null;
  }
}